Dependency notes

llf5185 opened the pull request:
Created documentation for node dependencies in the project.

llf5185 committed:
Added notes about dependencies to README.md

llf5185 committed:
Cleaned up Dependencies section in main README doc

Formatted the information to be more presentable and fixed some mistakes
in the writing.

llf5185 committed:
Documented root and server dependencies

llf5185 committed:
Fixed inaccuracy in the dependency notes

llf5185 committed:
Added description of prop-types to readme

Rtyujklop commented on README.md:
```diff
@@ -87,3 +87,92 @@ After sshing into the server, cd into either prod or test project. Then run resp
+
+## Dependencies
+### root
+`ajv`: seems to be unused, may be a subdependency. Compiles JSON schemas to JavaScript code.
```
Since ajv is flagged as “seems to be unused”, it might be worth explicitly noting whether it appears in dependencies vs devDependencies in package.json

llf5185 replied:
Changes from a recent commit now note which of the packages are used as devDependencies.

Rtyujklop commented on README.md:
```diff
@@ -87,3 +87,92 @@ After sshing into the server, cd into either prod or test project. Then run resp
+
+`@semantic-ui-react/css-patch`: patches semicolon issue with semantic (should have been fixed in an update of semanticUI?).
+
+`@testing-library/jest-dom`: currently unused.
```
For @testing-library/jest-dom and @testing-library/user-event, it might be helpful to say “kept for future test expansion” or “candidate for removal” so it’s clear whether they’re intentionally retained.

llf5185 replied:
A notice to keep these dependencies for now, has since been included.

Rtyujklop commented on README.md:
```diff
@@ -87,3 +87,92 @@ After sshing into the server, cd into either prod or test project. Then run resp
+
+`html-entities`: is used for the same thing as `he`.
+
+`prop-types`: a tool used for validating the data types of properties.
```
Maybe mention whether the project is using PropTypes consistently in new components, or if this is mainly for legacy components.

llf5185 replied:
I've now added more information about the components in which prop-types is used.

Rtyujklop commented on README.md:
```diff
@@ -87,3 +87,92 @@ After sshing into the server, cd into either prod or test project. Then run resp
+
+`html-to-text`: converts HTML into formatted text. Unclear if this needs to be in root.
+
+### server
```
This is a really helpful addition to the README – having dependency intent documented will make future upgrades and removals much safer.

Rtyujklop reviewed (commented):
These are the changes I can see now, overall it's good some tweaks can be done if you want.

llf5185 committed:
Referenced devDependencies and revised ajv info

Dependency documentation now notes which dependencies are
in devDependencies. Additionally, the accuracy of the documentation for
the package ajv has been improved.

llf5185 committed:
Merge branch 'dev' of github.com:RIT-Software-Engineering/RIT-SE-Senior-Project into dependency-notes

llf5185 committed:
Revised caniuse-lite documentation

Finished incomplete sentence in the caniuse-lite documentation and noted
that the package is a transitive dependency of react-scripts similar to ajv.

llf5185 committed:
Added reasoning for keeping unused testing packages

llf5185 committed:
Expanded documentation on dependency prop-types

Marrufof reviewed (approved):
It nice to have these written out.
